hummus doesn't build on node.js 4.0 :-(

As the subject says, I tried to install with node.js 4.0 (the new stable release) and it didn't work; though y'all should know

      CXX(target) Release/obj.target/hummus/src/hummus.o
    ../src/hummus.cpp:124:26: error: call to member function 'StartPDF' is ambiguous
            status = driver->StartPDF(*String::Utf8Value(args[0]->ToString()), pdfVersion,logConfig,PDFCreationSettings(compressStr...
                     ~~~~~~~~^~~~~~~~
    ../src/PDFWriterDriver.h:64:28: note: candidate function
        PDFHummus::EStatusCode StartPDF(const std::string& inOutputFilePath,
                               ^
    ../src/PDFWriterDriver.h:68:28: note: candidate function
        PDFHummus::EStatusCode StartPDF(v8::Handle<v8::Object> inStreamObject,
                               ^
    ../src/hummus.cpp:311:26: error: call to member function 'StartPDFParsing' is ambiguous
            status = driver->StartPDFParsing(*String::Utf8Value(args[0]->ToString()));
                     ~~~~~~~~^~~~~~~~~~~~~~~
    ../src/PDFReaderDriver.h:40:28: note: candidate function
        PDFHummus::EStatusCode StartPDFParsing(const std::string& inParsedFilePath);
                               ^
    ../src/PDFReaderDriver.h:41:28: note: candidate function
        PDFHummus::EStatusCode StartPDFParsing(v8::Handle<v8::Object> inStreamObject);
                               ^
    2 errors generated.
    make: *** [Release/obj.target/hummus/src/hummus.o] Error 1
    gyp ERR! build error
    gyp ERR! stack Error: `make` failed with exit code: 2
    gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
    gyp ERR! stack     at emitTwo (events.js:87:13)
    gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
    gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
    gyp ERR! System Darwin 14.5.0
    gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
    gyp ERR! cwd /Users/richard/code/hamstudy/node_modules/hummus
    gyp ERR! node -v v4.0.0
    gyp ERR! node-gyp -v v3.0.1
    gyp ERR! not ok
    npm ERR! Darwin 14.5.0
    npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "--save" "hummus@1.0.41"
    npm ERR! node v4.0.0
    npm ERR! npm  v2.14.2
    npm ERR! code ELIFECYCLE

    npm ERR! hummus@1.0.41 install: `node-gyp rebuild`
    npm ERR! Exit status 1
    npm ERR!
    npm ERR! Failed at the hummus@1.0.41 install script 'node-gyp rebuild'.
    npm ERR! This is most likely a problem with the hummus package,
    npm ERR! not with npm itself.
    npm ERR! Tell the author that this fails on your system:
    npm ERR!     node-gyp rebuild
    npm ERR! You can get their info via:
    npm ERR!     npm owner ls hummus
    npm ERR! There is likely additional logging output above.

--- a/src/hummus.cpp
+++ b/src/hummus.cpp
@@ -121,7 +121,7 @@ METHOD_RETURN_TYPE CreateWriter(const ARGS_TYPE& args)
     }
     else
     {
-        status = driver->StartPDF(*String::Utf8Value(args[0]->ToString()), pdfVersion,logConfig,PDFCreationSettings(compressStreams,true));
+        status = driver->StartPDF(std::string(*String::Utf8Value(args[0]->ToString())), pdfVersion,logConfig,PDFCreationSettings(compressStreams,true));
     }
     
     if(status != PDFHummus::eSuccess)
@@ -308,8 +308,8 @@ METHOD_RETURN_TYPE CreateReader(const ARGS_TYPE& args)
     if(args[0]->IsObject())
         status = driver->StartPDFParsing(args[0]->ToObject());
     else
-        status = driver->StartPDFParsing(*String::Utf8Value(args[0]->ToString()));
         
+        status = driver->StartPDFParsing(std::string(*String::Utf8Value(args[0]->ToString())));
     if(status != PDFHummus::eSuccess)
     {
 		THROW_EXCEPTION("Unable to start parsing PDF file");

